### PR TITLE
Use RequireSiteExists annotation to check site

### PR DESCRIPTION
**Pull request by phuongnq**
https://github.com/craftercms/craftercms/issues/6397

* Use `RequireSiteExists` annotation instead of `siteService.checkSiteExists(site);`
* When a method has `RequireSiteReady`, there is no need to further check site exists since site ready has also check if the site first exists.

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v2/service/content/ContentServiceImpl.java [hunk 1]
```diff
@@ -96,23 +97,23 @@ public class ContentServiceImpl implements ContentService, ApplicationContextAwa
     private org.craftercms.studio.api.v1.service.content.ContentService contentServiceV1;
 
     @Override
+    @RequireSiteExists
     @HasPermission(type = DefaultPermission.class, action = PERMISSION_CONTENT_READ)
-    public boolean contentExists(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId,
+    public boolean contentExists(@ProtectedResourceId(SITE_ID_RESOURCE_ID) @SiteId String siteId,
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/content/ContentServiceImpl.java]: `SiteId` itself is annotated with `@ProtectedResourceId`. Please validate just `@SiteId` would be enough

File: src/main/java/org/craftercms/studio/impl/v2/service/dependency/DependencyServiceImpl.java [hunk 2]
```diff
@@ -79,14 +75,13 @@ public List<String> getDependentPaths(@SiteId String siteId,
...
     public List<String> getHardDependencies(@SiteId String site,
                                             @ProtectedResourceId(PATH_LIST_RESOURCE_ID) List<String> paths) throws ServiceLayerException {
-        siteService.checkSiteExists(site);
         return dependencyServiceInternal.getHardDependencies(site, paths);
     }
 
+    @RequireSiteExists
     @HasPermission(type = DefaultPermission.class, action = PERMISSION_CONTENT_READ)
-    public List<DependencyItem> getDependentItems(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId,
+    public List<DependencyItem> getDependentItems(@SiteId @ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId,
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/dependency/DependencyServiceImpl.java]: @SiteId itself is annotated with @ProtectedResourceId

File: src/main/java/org/craftercms/studio/impl/v2/service/site/SitesServiceImpl.java [hunk 3]
```diff
@@ -85,14 +86,13 @@ public void updateSite(@SiteId String siteId, String name, String description)
...
-        checkSiteExists(siteId);
         sitesServiceInternal.updateSite(siteId, name, description);
     }
 
     @Override
+    @RequireSiteExists
     @HasPermission(type = DefaultPermission.class, action = PERMISSION_DELETE_SITE)
-    public void deleteSite(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId) throws ServiceLayerException {
-        checkSiteExists(siteId);
+    public void deleteSite(@SiteId @ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId) throws ServiceLayerException {
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/site/SitesServiceImpl.java]: SiteId itself is annotated with @ProtectedResourceId. Please validate just @SiteId would be enough

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/content/ContentServiceImpl.java (hunk 1 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/content/ContentServiceImpl.java]: Updated to have `@SiteId` only

File: src/main/java/org/craftercms/studio/impl/v2/service/dependency/DependencyServiceImpl.java (hunk 2 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/dependency/DependencyServiceImpl.java]: Updated to have `@SiteId` only

File: src/main/java/org/craftercms/studio/impl/v2/service/site/SitesServiceImpl.java (hunk 3 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/site/SitesServiceImpl.java]: Updated to have `@SiteId` only

**Review by jmendeza — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**